pkg/apis/events/validation/validation.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/api/core/v1"
+	apimachineryvalidation "k8s.io/apimachinery/pkg/api/validation"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/kubernetes/pkg/apis/core"
+	apivalidation "k8s.io/kubernetes/pkg/apis/core/validation"
+)
+
+const (
+	reportingInstanceLengthLimit = 128
+	actionLengthLimit            = 128
+	reasonLengthLimit            = 128
+	noteLengthLimit              = 1024
+)
+
+// ValidateEvent validates new created events.
+func ValidateEvent(event *core.Event) field.ErrorList {
+	allErrs := apivalidation.ValidateObjectMeta(&event.ObjectMeta, true, apimachineryvalidation.NameIsDNSSubdomain, field.NewPath("metadata"))
+	allErrs = append(allErrs, validateEventSpec(event)...)
+	return allErrs
+}
+
+// ValidateEventUpdate validates updated events.
+func ValidateEventUpdate(newEvent, oldEvent *core.Event) field.ErrorList {
+	allErrs := apivalidation.ValidateObjectMetaUpdate(&newEvent.ObjectMeta, &oldEvent.ObjectMeta, field.NewPath("metadata"))
+	allErrs = append(allErrs, validateEventSpec(newEvent)...)
```

Comment:
I'm wondering if we shouldn't be more strict with this validation and enforce that some fields don't change (e.g. ReportingController, ReportingInstance, Action, ...).

@liggitt - WDYT?

Reply:
@wojtek-t - +1 these fields should be immutable:

- the Event API represent a piece of information regarding something, I don't see why we would alter it
- on the client-side it's already impossible to alter these information

Reply:
Make sense. IIUC other than `Series` field, all remaining fields cannot be changed. My questions is whether to check all of them or just some important fields?

Reply:
Let's add a function and start validating those - we will figure out those during the review.
I would say let's validate everything we can but we can wait for @liggitt with it).

Reply:
I've add a check for everything except series field.